### ISSUE TITLE
9. Accessibility

### ISSUE THREAD

**Issue by marcoscaceres** [w3c/dpub-pwp-ucr#133]
> 9.1 Accessible Metadata

Having metadata about this seems unhelpful. 

For example:

> Ferdous wants to buy a book about a museum exhibit and wants to guarantee before he does that his screen reader will describe the images to him in great detail.

As formulated, the requirement here is on the screen reader - not that the images are described by the author. Also, it's misguided in that it assumes that the images are described independently of the surrounding text.

> Alejandra must ensure that the equation editors in her textbook have non-mouse functionality before she purchases the textbook.
- "Said no one ever".  Seriously, how one would even express this or what this even means makes no sense. Will Alejandra assert, "well, I have to be able to click and double click or it's a deal breaker for me!?". 

> A university professor is developing a course and the professor knows that he is required to use accessible digital materials. The professor uses the search capabilities of available publications to determine which titles are accessible and therefore suitable for his use.

This requirement makes no sense either: "accessibility" is not checkbox or signal - it's an emergent quality of the text, based on semantic structure, and annotations. That the Professor could just search for "accessible things" doesn't make any sense.

> Anita is a school student who knows only uncontracted Hindi Braille. Her parents should be able to discover the language and the braille codes of the publications before purchasing the publications for her.

Again, this is independent of the text. A store may provide this information as an aside to the text.

> Jessie is taking a class and the professor has assigned Chapter 5 in an art textbook for review, but before Jessie purchases the book she wants to ensure that Chapter 5 is accessible.

As above. "Accessible" is not a checkbox - it's not something that can be assured via some magical metadata.

**Comment by iherman** [comment]
This is a much more general discussion, and I believe many people will disagree with your characterization as being "unhelpful"...

There is currently work going on at schema.org to include exactly those metadata in the the schema.org hierarchy. AFAIK, the last steps to conclude this were made at TPAC.

See:
- https://github.com/daisy/epub-revision-a11y/wiki/ePub-3.1-Accessibility--Proposal-To-Schema.org
- https://github.com/schemaorg/schemaorg/issues/1100

That being said, if those terms are indeed included in the schema.org hierarchy, this document may simply refer to the fact that those metadata should be expressible for a WP, and it is not necessary to include a full section for those here.

**Comment by TzviyaSiegman** [comment]
> "Said no one ever". Seriously, how one would even express this or what this even means makes no sense. Will Alejandra assert, "well, I have to be able to click and double click or it's a deal breaker for me!?".

Based on the work I've done with the DAISY Consortium, this is a common request especially from those purchasing textbooks for institutions that are legally mandated to supply accessible materials. See http://pending.schema.org/ for work in progress. Additional terms are in progress. We expect this to be in the next draft of schema.org. See also https://github.com/schemaorg/schemaorg/issues/1100 for details of how this can be expressed in a meaningful way.

**Comment by rdeltour** [comment]
@marcoscaceres:

> As formulated, the requirement here is on the screen reader - not that the images are described by the author. Also, it's misguided in that it assumes that the images are described independently of the surrounding text.

Proposed rewording:

> Ferdous wants to buy a book about a museum exhibit and wants to guarantee before he does that images are associated with detailed descriptive text (to ensure his screen reader will be able to describe the images to him in great detail).

@marcoscaceres:

> "Said no one ever". Seriously, how one would even express this or what this even means makes no sense. Will Alejandra assert, "well, I have to be able to click and double click or it's a deal breaker for me!?". 

The wording does sound confusing. AFAIU this is about ensuring the UI is accessible to people who can't use a mouse. How about:

Proposed rewording:

> Alejandra cannot use a mouse and needs to ensure that the equation editors in her textbook is operable with a keyboard before she purchases the textbook.

@marcoscaceres:

> This requirement makes no sense either: "accessibility" is not checkbox or signal

I believe this is about discovering whether some content complies to some accessibility guidelines or standards (e.g. WCAG 2.0 AA, section 508, French RGAA 3.0). 

Proposed rewording:

> A university professor is developing a course and the professor knows that he is required by the university's policy to use digital materials that conform to WCAG 2.0 level AA. The professor uses the search capabilities of available publications to determine which titles are conformant and therefore suitable for his use.

@marcoscaceres:

> Again, this is independent of the text. A store may provide this information as an aside to the text.

Well, that's some definition of "metadata", no? ;-)
The associated requirement would be that the author must be able to associate some metadata to the content.

> As above. "Accessible" is not a checkbox - it's not something that can be assured via some magical metadata. 

Right, and right. But "conformant to accessibility specification XXX" _can_ be evaluated, and "certified conformant to accessibility specification XXX by organization YYY" is a property that can certainly be declared in metadata.

Proposed rewording:

> Jessie is taking a class and the professor has assigned Chapter 5 in an art textbook for review, but, as a low-vision user, Jessie  wants to ensure that the accessibility features of Chapter 5 have been certified by a trusted organization before purchases the book.

**Comment by clapierre** [comment]
we will be discussing this during our DPUB A11Y call this Friday.  
Thanks Romain for your suggestions!

**Comment by marcoscaceres** [comment]
@TzviyaSiegman, I understand the wider requirement. But what is a concrete example of an "equation editor"? That's the bit that bugs me because it's just an application of HTML or, at worst, some javascript trying to hack together some MathML. 

Maybe I never got the memo about the web being able to do Math, let alone edit it inline in a web app? I'd love to see an example.

**Comment by avneeshsingh** [comment]
Romain:
Ferdous wants to buy a book about a museum exhibit and wants to guarantee before he does that images are associated with detailed descriptive text (to ensure his screen reader will be able to describe the images to him in great detail).

Avneesh:
I will recommend to generalize the screen reader instead of placing it in parenthesis.
"Ferdous wants to buy a book about a museum exhibit, but before he does that,  he wants to guarantee that the images are associated with detailed text descriptions, to ensure that he will be able to understand it with a screen reader or refreshable Braille display."

Romain:
Alejandra cannot use a mouse and needs to ensure that the equation editors in her textbook is operable with a keyboard before she purchases the textbook.

Avneesh: Looks good.

Romain:
A university professor is developing a course and the professor knows that he is required by the university's policy to use digital materials that conform to WCAG 2.0 level AA. The professor uses the search capabilities of available publications to determine which titles are conformant and therefore suitable for his use.

Avneesh:
Your logic looks good. I am confused about the following line, it looks that there is a missing piece here.
"The professor uses the search capabilities of available publications to determine which titles are conformant and therefore suitable for his use."

Romain:
Jessie is taking a class and the professor has assigned Chapter 5 in an art textbook for review, but, as a low-vision user, Jessie wants to ensure that the accessibility features of Chapter 5 have been certified by a trusted organization before purchases the book.

Avneesh:
We should also have a use case that is not focused on certified content, there may be content that does not conform to WCAG, but can have accessibility features to satisfy the needs of a segment of users.
"Jessie is taking a class and the professor has assigned Chapter 5 in an art textbook for review, but, as a low-vision user, Jessie wants to ensure that the chapter has the accessibility features that she needs."

**Comment by avneeshsingh** [comment]
I Missed out replying to comments for this one. I should clarify that braille code are for text. This information about encoding has to be available for search and discovery at the retail store, and also to the reader of the Braille publication. So, it cannot be addressed by only external metadata records.

"Anita is a school student who knows only uncontracted Hindi Braille. Her parents should be able to discover the language and the braille codes of the publications before purchasing the publications for her."

**Comment by marcoscaceres** [comment]
> On 1 Oct. 2016, at 2:23 am, Avneesh Singh notifications@github.com wrote:
> 
> I Missed out replying to comments for this one. I should clarify that braille code is encoding of the text. This information about encoding has to be available for search and discovery at the retail store, and also to the reader of the Braille publication. So, it cannot be addressed by only external metadata records.
> 
> "Anita is a school student who knows only uncontracted Hindi Braille. Her parents should be able to discover the language and the braille codes of the publications before purchasing the publications for her."

Much better. It doesn't make a requirement presupposed on a technology or solution.

**Comment by avneeshsingh** [comment]
It is a usecase for metadata for braille codes.

**Comment by lrosenthol** [comment]
These changes are incorporated into the latest draft

**Comment by clapierre** [comment]
ACTION: Charles will ensure all 4 updates to the various use-cases have been brought into the updated de-duped copy.  

The Ferdous use-case has already been updated in the Repo.

Below are all 4 with final tweaks from todays A11Y TF meeting.

Alejandra cannot use a mouse and needs to ensure that the equation editors in her textbook is operable with a keyboard before she purchases the textbook.

A university professor is developing a course and the professor knows that he is required by the university's policy to use digital materials that conform to WCAG 2.0 level AA. The professor searches to determine which titles are accessible and therefore suitable for his use.

Jessie is taking a class and the professor has assigned Chapter 5 in an art textbook for review, but, as a low-vision user, Jessie wants to ensure that the chapter has the accessibility features that she needs.

Anita is a school student who knows only uncontracted Hindi Braille. Her parents should be able to discover the language and the braille codes of the publications before purchasing the publications for her.

**Comment by iherman** [comment]
Part of the latest version #157
